auth/store.go

```diff
@@ -124,6 +127,40 @@ func (as *authStore) UserDelete(r *pb.AuthUserDeleteRequest) (*pb.AuthUserDelete
 	return &pb.AuthUserDeleteResponse{}, nil
 }
 
+func (as *authStore) UserChangePassword(r *pb.AuthUserChangePasswordRequest) (*pb.AuthUserChangePasswordResponse, error) {
+	hashed, err := bcrypt.GenerateFromPassword([]byte(r.Password), bcrypt.DefaultCost)
```

Comment:
I am not sure if we can do this at api layer. once this goes through raft, the requests will be serialized. so it blocks other requests. and bcrypt takes a long time to finish. i am a little bit worry about that. but we can deal with this later for sure.